### ISSUE TITLE
add robot state

### ISSUE THREAD

**Issue by guoyanzi** [InternRobotics/InternVLA-M1#24]
I want to include state information in my training.
Currently, the actions I train on are the arm joint angles, and the state is also the arm joint angles.
The gripper state is either 0 or 1.

But after I added the state into training, during testing I found that the predicted gripper actions fluctuate a lot — it often switches rapidly, grabbing and then immediately releasing.
Here is the code I modified.
In the model definition, I added:
`self.proprio_encoder = nn.Linear(8, 2048)
`
in forward:
```
device = next(self.proprio_encoder.parameters()).device
proprios = torch.tensor(np.array(state), dtype=torch.float16).to(device)
proprio_embeds = self.proprio_encoder(proprios)

start_layer = self.config.framework.layer_qformer.qformer_start_layer
end_layer = self.config.framework.layer_qformer.qformer_end_layer
condition_features = qwenvl_outputs.hidden_states[start_layer:end_layer]
cat_conditions = []
for layer_index in range(len(condition_features)):
    layer_features = condition_features[layer_index]  # [B, n_qformer_token, D]
    layer_features = torch.cat(
        [layer_features, dino_encoded_features,proprio_embeds], dim=1
    )  # [B, n_qformer_token + num_view * token, D]
    cat_conditions.append(layer_features)

start_time_action_pre = time.time()
action_condition_feature = self.layer_qformer(cat_conditions)  # [B, 64, D_action]
```
Can you help review this and give some guidance?

**Comment by Yioutpi** [comment]
Thanks for sharing the details. You may try the following:

i) Check whether the state uses the same normalization in training and inference, and make sure predict_action also receives the state during inference.
ii) If you have a wrist view, consider removing the gripper state and only using joint angles as input.
iii) If you keep the gripper state, verify that its normalized range is compatible with joint angles.
